test/Math/max2.js

```diff
@@ -0,0 +1,25 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Copyright (c) ChakraCore Project Contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+let calledValueOf = false;

```

Comment:
I think to hit all the possible paths through the code above we should add tests with 4 parameters one for NaN in each spot. AND for each test should verify that NaN is returned (use the isNaN method - or test for strict equality with itself whichever you think is neater) as well as verifying that valueOf has been called the correct number of times (use an increment)